pmlc/rt/vulkan/tests/edsl_test_gpu.cc

```diff
@@ -0,0 +1,937 @@
+// Copyright 2020 Intel Corporation
+// RUN: cc_test_gpu | FileCheck %s
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <random>
+
+#include "llvm/ADT/StringRef.h"
+
+#include "plaidml/edsl/autodiff.h"
+#include "plaidml/edsl/edsl.h"
+#include "plaidml/exec/exec.h"
+#include "plaidml/op/op.h"
+#include "plaidml/testenv.h"
+#include "pmlc/util/env.h"
+#include "pmlc/util/logging.h"
+
+using llvm::StringRef;
+using ::testing::ContainerEq;
+using ::testing::Eq;
+
+namespace plaidml::edsl {
+namespace {
+
+class CppEdsl : public TestFixture {};
+
+Program makeProgram(const std::string &name,
+                    const std::vector<Tensor> &outputs) {
+  auto device = plaidml::Settings::get("PLAIDML_DEVICE");
+  auto target = plaidml::Settings::get("PLAIDML_TARGET");
+
+  plaidml::Settings::set("PLAIDML_DEVICE", "intel_gen");
+  plaidml::Settings::set("PLAIDML_TARGET", "intel_gen");
+  auto program = ProgramBuilder(name, outputs).compile();
+  std::cout << program << std::endl;
+  plaidml::Settings::set("PLAIDML_DEVICE", device);
+  plaidml::Settings::set("PLAIDML_TARGET", target);
+  return program;
+}
+
+Tensor Dot(const Tensor &X, const Tensor &Y) {
+  TensorDim I, J, K;
+  TensorIndex i("i"), j("j"), k("k");
+  X.bind_dims(I, K);
+  Y.bind_dims(K, J);
+  auto R = TensorOutput(I, J);
+  R(i, j) += X(i, k) * Y(k, j);
+  return R;
+}
+/*
```

Comment:
Remove all these commented out tests, we can re-add them when they start working as needed.

Reply:
Ah, I see. I will remove them.